plugins/jobs/web_client/views/JobStatusSegmentizer.js

```diff
@@ -0,0 +1,37 @@
+import _ from 'underscore';
+import JobStatus from '../JobStatus';
+
+export default class JobStatusSegmentizer {
+    segmentize(jobs) {
```

Comment:
The logic is extracted let's keep it here for now and extract it to other places when the logic is useful elsewhere.